reactive/flannel.py

```diff
@@ -243,7 +244,19 @@ def ensure_etcd_connections():
     relevant flags to make sure accurate config is regenerated.
     '''
     etcd = endpoint_from_flag('etcd.available')
-    if data_changed('flannel_etcd_connections', etcd.get_connection_string()):
+    connection_changed = data_changed('flannel_etcd_connections',
+                                      etcd.get_connection_string())
+    cert_changed = data_changed('flannel_etcd_cert',
+                                etcd.get_client_credentials())
+    if connection_changed or cert_changed:
+        etcd.save_client_credentials(ETCD_KEY_PATH,
+                                     ETCD_CERT_PATH,
+                                     ETCD_CA_PATH)
+        clear_flag('flannel.service.installed')
+
+        # Clearing the above flag will change config that the flannel
+        # service depends on. Set ourselves up to (re)invoke the start handler.
+        clear_flag('flannel.service.started')
```

Comment:
Looks like the clear_flags here got duplicated. I'll push up a cleanup commit in a bit here